### DISCUSSION TITLE
Mute on entry

### DISCUSSION THREAD

**Discussion by reinhardwh** [jamulussoftware/jamulus#2520] [Ideas]
Would be great to offer

**"Muted entry"** for new participants in server settings

worldjam.vip recommend this

Or set as option for client

**Comment by dcorson-ticino-com** [comment]
Muting is not a server function.
You can set New Client Level to 0 if you don't want to hear new participants.
I would not want to because it would mean I need to pull up the level of everyone coming into a session.

EDIT: I do not see any mention of muting new participants on the worldjam.vip site.  Where did you see that?

**Reply by reinhardwh** [reply]
Then it would be good to offer in client. 
Similar to ZOOM
This avoid to disturb sessions when entering.

**Reply by pljones** [reply]
Anyone can choose to enter a session with Mute Myself enabled already.

**Reply by dcorson-ticino-com** [reply]
If you don't play or talk when you enter a session there will be no disturbance, no matter how the settings are.
In any case in comparison with ZOOM everything is already there, what do you need more?

**Reply by reinhardwh** [reply]
Thats right - but sometimes - people forget...
Maybe good if after starting the client the "automute" is active (as a option)

**Reply by dcorson-ticino-com** [reply]
> Thats right - but sometimes - people forget... Maybe good if after starting the client the "automute" is active (as a option)

The problem we see more often is that people don't realize they are muted and then they are missing in the mix.

**Reply by pcar75** [reply]
> Muting is not a server function.

?!?! How can Muting (channels, not Mute Myself) not be a server function. Though, it is set by a (client) user muting/panning his or other(s) channels, it is definitely the server that is adjusting the mix for that user.  My understanding is therefore that the server integrates and resends not individual tracks but personnalized mix to each and every participants ... the server does the job.  Which may explain the wish : for large ensembles and choirs to send one same standard mix to each user (thus lessening the power/performance requirement on the server).

**Reply by dcorson-ticino-com** [reply]
The server does not know if someone is muted or not soloed or if the volume is just turned down.
The server mixes, but has no idea of why the volume multiplier is 0.

**Reply by pcar75** [reply]
EDIT :  Apologies to any inconvenienced, it was not my intent to derail this topic.
I've opened "Better understanding Client vs. Server, #2530 " in the General category.

**Comment by ann0see** [comment]
I think I get your point: you basically want that everyone who joins is first muted to not disturb the session. I think a CLI option already exists. 

Moreover, I think this has been discussed before but rejected by Volker/Community (?) due to the argument that it might cause confusion. 

I think the real question is: why do you want to have everyone having mute myself on before joining? 
* To ensure their audio setup is correct before joining —> implement something like a pre join sound check
* To prevent disturbances —> feedback protection might be improved
* Because Zoom does it (Jamulus should have basic zoom features) —> ???

**Reply by reinhardwh** [reply]
Hello @ann0see,

Its more a "fail safe" option.

In case musicians would like to join new sessions "muted"

Its not a copy of ZOOM function but a standard/option also in other conferencing SW like MSFT Teams.

**Reply by ann0see** [reply]
> Its more a "fail safe" option.
> In case musicians would like to join new sessions "muted"

But can we assume everyone wants to join muted? 
I think we could add a switch similar to the `--mutestream` option in the GUI options?

**Reply by reinhardwh** [reply]
Excellent, yes.

This option is a bit "hidden" for normal Jamulus musicians

https://jamulus.io/de/wiki/Command-Line-Options

Client only commands
-M or --mutestream Starts Jamulus in muted state

To offer this option in Jamulus settings tab would be helpful

![image](https://user-images.githubusercontent.com/77500932/159114525-d5ad41e9-1acc-40b0-a6cb-89880f58d622.jpeg)

**Reply by pljones** [reply]
This doesn't need adding to the UI anywhere.  For UI usage, all that would change would be to store the existing UI value.

Of course, turning the setting off in the UI would mean the next run, it would be off.

Hence why the command line option is there: it means "Run Jamulus with mute myself".  It makes far more sense than storing the UI value as a setting.

**Reply by reinhardwh** [reply]
I like the "command line option".

On the other hand:
Not everybody 
- read the guidelines
- knows how to create shortcuts and add CLI commands

A UI would make it easier to use the feature :)

There is also one main difference to CLI option:

Enter _new_ sessions muted is not supported right now

I consider this a useful feature

**Reply by dcorson-ticino-com** [reply]
We should never automatically mute.  
I can see all the complaints that Jamulus doesn't work here in the discussion already.  
You can mute, but it must be a conscious action.  And even then many forget, as we know.
This was a potential problem before we had the feedback protection, but now no longer.
And even before the feedback protection we saw more problems with people inadvertently muting than crashing a session with noise.
If this really was a problem we could add a window at startup, like Zoom does, to make this a conscious action.  We could maybe use the command line option to enable that. 
But although this was a potential problem before feedback protection, it is now no more.

**Reply by reinhardwh** [reply]
Ok, thanks for the discussion.

I understand your concerns and trust your experience.

**Reply by rdica** [reply]
If a user is going to go to the lengths of adding a cli option, they can more easily "Mute Yourself" before they join a session. Starting up the client doesnt automatically join a session, so what problem gets solved by this suggestion?

**Reply by pcar75** [reply]
> You can mute, (...) And even then many forget, as we know.
> (...) And even before the feedback protection we saw more problems with people inadvertently muting than crashing a session with noise.

Often, a session is paused to figure out the problems of someone and find that there is no problem just that one forgot to un'Mute Myself'. 
EDIT : Apologies to any inconvenienced, it was not my intent to derail this topic.
I've added my comment to "Need Indication On Fader Panel That Someone Has Muted Themselves" #1025  UI Related category.

**Reply by wschell** [reply]
In my opinion it wouldn't help much if it is a client option. If I think of setting such an option I also think of mute myself before entering a server. I would vote for a server option - so the server tells the client to set the option on connect. So the server operator decides if people can only connect self-muted.

**Reply by reinhardwh** [reply]
Like the idea. 
Some kind of "Open Stage Rule"

In server welcome message a info could be shown by server admin.

**Comment by pcar75** [comment]
I would strongly suggest a documentation section How to Use Jamulus ... recommending that users should first try their sound with loopback address before joining a public session just to make sure that they don't cause other musicians to stop and mute any ear-splitting noise they are generating !

Sadly, that may not prevent careless or *\!&#"/?ole users from ruining a session.

**Reply by ann0see** [reply]
Can you draft a suggestion and put it on the GettingStarted page - if it isn't there already which I doubt. 
Maybe a loopback check should also be implemented https://github.com/jamulussoftware/jamulus/discussions/1252

It would be great if you helped re-vive it (or draft an actionable issue)

**Reply by pcar75** [reply]
Oops sorry I had missed that

> Before playing with others, we recommend that you connect to an empty server to test your setup, and make sure you are listening to the signal coming back from the server (if possible) and not yourself.

The loopback test bit could be added to this phrase but may burden the idea with unnecessary details.  However, it might be more complete if 
> Before playing with others, we recommend that you connect to an empty server to test your setup, and make sure you are listening to the signal coming back from the server (if possible) and not yourself **and that your signal has appropriate quality and volume.**

For loopback check to be (semi?-) automatically implemented , one should be warned the ping and overall delays do not reflect what a public or private server elsewhere will provide, as evidenced in Ref. "3.8.0 on Windows 10 : problem with loopback (127.0.0.1)" https://github.com/jamulussoftware/jamulus/discussions/1855

**Reply by dcorson-ticino-com** [reply]
If people follow the getting started info it will not happen.
But if it is not clear there, suggestions for clarification are always welcome.

**Reply by pcar75** [reply]
@dcorson-ticino-com  Sadly , anyone working around adult education will testify that contrary to what we tell our children to do, most adult do not read the manuals :-/
EDIT : ... but some may skim over a Getting Started section .
